test/Microsoft.AspNet.WebUtilities.Tests/HttpResponseStreamWriterTest.cs

```diff
@@ -0,0 +1,461 @@
+            char character,
+            int charCount,
+            string encodingName)
+        {
+            // Arrange
+            var encoding = Encoding.GetEncoding(encodingName);
+            string data = new string(character, charCount);
+            var expectedBytes = encoding.GetBytes(data);
+            var stream = new MemoryStream();
+            var writer = new HttpResponseStreamWriter(stream, encoding);
+
+            // Act
+            using (writer)
+            {
+                await writer.WriteAsync(data);
+            }
+
+            // Assert
+            Assert.Equal(expectedBytes, stream.ToArray());
+        }
+
+        // None of the code in HttpResponseStreamWriter differs significantly when using pooled buffers.
+        //
+        // This test effectively verifies that things are correctly constructed and disposed. Pooled buffers
+        // throw on the finalizer thread if not disposed, so that's why it's complicated.
+        [Fact]
+        public void HttpResponseStreamWriter_UsingPooledBuffers()
+        {
+            // Arrange
+            var encoding = Encoding.UTF8;
+            var stream = new MemoryStream();
+
+            var expectedBytes = encoding.GetBytes("Hello, World!");
+
+            using (var writer = new HttpResponseStreamWriter(
+                stream,
+                encoding,
+                1024,
+                ArrayPool<byte>.Shared,
+                ArrayPool<char>.Shared))
+            {
+                // Act
+                writer.Write("Hello, World!");
+            }
+
+            // Assert
+            Assert.Equal(expectedBytes, stream.ToArray());
+        }
+
+        private class TestMemoryStream : MemoryStream
+        {
+            public int FlushCallCount { get; private set; }
+
+            public int FlushAsyncCallCount { get; private set; }
+
+            public int CloseCallCount { get; private set; }
+
+            public int DisposeCallCount { get; private set; }
+
+            public int WriteCallCount { get; private set; }
+
+            public int WriteAsyncCallCount { get; private set; }
+
+            public override void Flush()
+            {
+                FlushCallCount++;
+                base.Flush();
+            }
+
+            public override Task FlushAsync(CancellationToken cancellationToken)
+            {
+                FlushAsyncCallCount++;
+                return base.FlushAsync(cancellationToken);
+            }
+
+            public override void Write(byte[] buffer, int offset, int count)
+            {
+                WriteCallCount++;
+                base.Write(buffer, offset, count);
+            }
+
+            public override Task WriteAsync(byte[] buffer, int offset, int count, CancellationToken cancellationToken)
+            {
+                WriteAsyncCallCount++;
+                return base.WriteAsync(buffer, offset, count, cancellationToken);
+            }
+
+#if DNX451
```

Comment:
You're looking for net451 right? I suppose I should just replace this every time I see it in this repo?